packages/react-components/src/components/table-output-component.tsx

```diff
@@ -508,13 +516,28 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
         return lines[0].index;
     }
 
-    private async findNextMatchedEvent() {
+    private async findMatchedEvent(direction: Direction) {
         let isFound = false;
         if (this.gridApi) {
             this.gridApi.deselectAll();
+            let rowNodes: RowNode[] = [];
+            let currRowIndex = this.selectStartIndex === -1 ? this.gridApi?.getLastDisplayedRow() + 1 : this.selectStartIndex;
+            const blockSize = this.props.cacheBlockSize ? this.props.cacheBlockSize : 200;
             this.gridApi.forEachNode(rowNode => {
-                if (rowNode.rowIndex > Math.max(this.selectStartIndex, this.selectEndIndex) && rowNode.data
-                    && rowNode.data['isMatched'] && !isFound) {
+                // make sure to search only in current block and not in full cache
+                if (Math.floor(currRowIndex / blockSize) === Math.floor(rowNode.rowIndex / blockSize)) {
+                    rowNodes.push(rowNode);
+                }
+            });
+            if (direction === Direction.PREVIOUS) {
+                rowNodes = rowNodes.reverse();
+            }
+            rowNodes.forEach(rowNode => {
+                if ((rowNode.data && rowNode.data['isMatched'] && !isFound) &&
+                    (((direction === Direction.PREVIOUS) &&
+                        (rowNode.rowIndex < Math.max(this.selectStartIndex, this.selectEndIndex))) ||
```

Comment:
Will this prevent searching backwards if there is no selection (both are -1, right?)

Reply:
mhh, maybe it should compare with currRowIndex instead? I'll check